### PR TITLE
Add vnet linking for private dns

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###

[AB#816]

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: templates/arm-vnet-template.json [hunk 1]
```diff
@@ -438,6 +442,21 @@
...
         "[resourceId('Microsoft.Network/networkSecurityGroups',parameters('iaasNsgName'))]"
       ]
+    },
+    {
+      "type": "Microsoft.Network/privateDnsZones/virtualNetworkLinks",
+      "apiVersion": "2018-09-01",
+      "name": "[concat(variables('privateDnsZone')[copyIndex()].name, '/', parameters('aksVnetName') )]",
+      "location": "global",
+      "dependsOn": [
+          "[resourceId('Microsoft.Network/privateDnsZones', variables('privateDnsZone')[copyIndex()].name)]"
```
Comment (timja) [inline templates/arm-vnet-template.json]: don't you need to pass the resource group as well? if it's not in the same resource group?

Comment (luigibk) [reply templates/arm-vnet-template.json]: Good point but it doesn't look like I need that, judging by the exported template I am looking at.

File: templates/vars/vnet/sbox.json [hunk 2]
```diff
@@ -7,6 +7,9 @@
     },
     "aksVnetDnsServers": {
       "value": ["10.100.139.254"]
+    },
+    "privateDnsZones": {
+      "value": ["service.core-compute.sandbox.internal", "sandbox.platform.hmcts.net"]
```
Comment (timja) [inline templates/vars/vnet/sbox.json:12]: what happens if you run this and the list is empty?

Comment (timja) [inline templates/vars/vnet/sbox.json:12]: what about the idam zone?

**Pull request merged**